staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/stdlib.go

```diff
@@ -0,0 +1,585 @@
+	}
+	sz := lister.Size().Value().(int64)
+	for i := int64(0); i < sz; i++ {
+		if lister.Get(types.Int(i)).Equal(item) == types.True {
+			return types.Int(i)
+		}
+	}
+	return types.Int(-1)
+}
+
+func lastIndexOf(list ref.Val, item ref.Val) ref.Val {
+	lister, ok := list.(traits.Lister)
+	if !ok {
+		return types.NoSuchOverloadErr()
+	}
+	sz := lister.Size().Value().(int64)
+	for i := sz - 1; i >= 0; i-- {
+		if lister.Get(types.Int(i)).Equal(item) == types.True {
+			return types.Int(i)
+		}
+	}
+	return types.Int(-1)
+}
+
+func find(strVal ref.Val, regexVal ref.Val) ref.Val {
+	str, ok := strVal.Value().(string)
+	if !ok {
+		return types.NoSuchOverloadErr()
+	}
+	regex, ok := regexVal.Value().(string)
+	if !ok {
+		return types.NoSuchOverloadErr()
+	}
+	re, err := regexp.Compile(regex)
+	if err != nil {
+		return types.NewErr("Illegal regex: %v", err.Error())
+	}
+	result := re.FindString(str)
+	return types.String(result)
+}
+
+func findAll(args ...ref.Val) ref.Val {
+	argn := len(args)
+	if argn < 2 || argn > 3 {
+		return types.NoSuchOverloadErr()
+	}
+	str, ok := args[0].Value().(string)
+	if !ok {
+		return types.NoSuchOverloadErr()
+	}
+	regex, ok := args[1].Value().(string)
+	if !ok {
+		return types.NoSuchOverloadErr()
+	}
+	n := int64(-1)
+	if argn == 3 {
+		n, ok = args[2].Value().(int64)
+		if !ok {
+			return types.NoSuchOverloadErr()
+		}
+	}
+
+	re, err := regexp.Compile(regex)
+	if err != nil {
+		return types.NewErr("Illegal regex: %v", err.Error())
+	}
+
+	result := re.FindAllString(str, int(n))
+
+	return types.NewStringList(types.DefaultTypeAdapter, result)
+}
+
+func stringToUrl(arg ref.Val) ref.Val {
+	s := arg.Value().(string)
+	// Use ParseRequestURI to check the URL before conversion.
+	// ParseRequestURI requires absolute URLs and is used by the OpenAPIv3 'uri' format.
+	_, err := url.ParseRequestURI(s)
+	if err != nil {
+		return types.NewErr("URL parse error during conversion from string: %v", err)
+	}
+	// We must parse again with Parse since ParseRequestURI incorrectly parses URLs that contain a fragment
+	// part and will incorrectly append the fragment to either the path or the query, depending on which it was adjacent to.
+	u, err := url.Parse(s)
+	if err != nil {
+		// Errors are not expected here since Parse is a more lenient parser than ParseRequestURI.
+		return types.NewErr("URL parse error during conversion from string: %v", err)
+	}
+	return model.URL{URL: u}
+}
+
+func getScheme(arg ref.Val) ref.Val {
+	u := arg.Value().(*url.URL)
+	return types.String(u.Scheme)
+}
+
+func getHost(arg ref.Val) ref.Val {
+	u := arg.Value().(*url.URL)
+	return types.String(u.Host)
+}
+
+func getHostname(arg ref.Val) ref.Val {
+	u := arg.Value().(*url.URL)
+	return types.String(u.Hostname())
+}
+
+func getPort(arg ref.Val) ref.Val {
+	u := arg.Value().(*url.URL)
+	return types.String(u.Port())
+}
+
+func getPath(arg ref.Val) ref.Val {
+	u := arg.Value().(*url.URL)
+	return types.String(u.EscapedPath())
```

Comment:
I think this makes sense, but it bothers me slightly that this isn't symmetric with getQuery (which returns unescaped keys/values)

Reply:
I've expanded the docs and examples and renamed `getPath` to `getEscapedPath`.